docs/protobuf-type-definitions.md

```diff
@@ -0,0 +1,48 @@
+# From Protobuf to TypeScript and Flow
+
+This document describes how the definitions of protobuf messages maintained in the
+firmware repo are semi-automatically translated to Flow and TypeScript definitions that are used in Connect codebase and by Connect users respectively.
+
+## The Pipeline
+
+The beginning and source of truth are the `.proto` definitions in the [firmware repository](https://github.com/trezor/trezor-firmware/tree/master/common/protob). These are duplicated as read-only in the [trezor-common](https://github.com/trezor/trezor-common) repository.
+
+`trezor-common` is included in Connect as a git submodule mounted at `submodules/trezor-common`.
+
+Here, `.proto` definitions are translated to a JSON format using [proto2js](https://www.npmjs.com/package/proto2js) package. This JSON is used on runtime by the [trezor-link](https://github.com/trezor/trezor-link/) (de)serialization logic and to generate the Flow and Typescript definitions.
+
+The JSON is transformed to Flow and/or TypeScript definitions by a script in `scripts/protobuf-types.js`. The script also applies 'patches' I.e. after-the-fact fixes manually described in `scripts/protobuf-patches.js`. The patches compensate for/fix
+- The source `.proto` definitions that do not reflect the actual business logic. Usually fields marked as required which are in fact optional.
+- Fields typed as `uint32`, `uint64`, `sint32`, `sint64` in protobuf that need to be represented as strings in runtime because of javascript number's insufficient range. Runtime conversion is handled automatically by `trezor-link`.
+- Similarly, fields typed as `bytes` in protobuf that are represented as base64 string in runtime.
```

Comment:
```suggestion
- Similarly, fields typed as `bytes` in protobuf may be represented as hexadecimal `string`, `Buffer`, `Uint8Array` or `Array<number>` in runtime.
```

see
https://github.com/trezor/connect/blob/develop/scripts/protobuf-patches.js#L111
or
https://github.com/trezor/connect/blob/develop/scripts/protobuf-patches.js#L133-L134
or
https://github.com/trezor/connect/blob/develop/scripts/protobuf-types.js#L20

Reply:
Ah, so this is even wider. Thanks for linking the examples!

Suggestion applied [here](https://github.com/trezor/connect/pull/785/commits/e8efb21fe44d752b3e76a151e7eb661a46502087).